### PR TITLE
Fix #704

### LINKED ISSUE #704: Killing with shotguns grants 1 kill bonus per bullet
In other words, one can get 30 chips for 1 kill, yikes.

### PR DESCRIPTION
**Pull request by Fueredoriku**

### CODE REVIEW

File: Assets/Scripts/Gamestate/PlayerManager.cs [hunk 1]
```diff
@@ -191,7 +191,8 @@ private void OnDeath(HealthController healthController, float damage, DamageInfo
         {
             killer = lastPlayerThatHitMe;
         }
-        onDeath?.Invoke(killer, this, info);
+        if (isAlive)
+            onDeath?.Invoke(killer, this, info);
```
Comment (toberge) [inline Assets/Scripts/Gamestate/PlayerManager.cs]: Note that by guarding only this invocation, you're not preventing the ridiculous amount of knockback (and multiple invocations of the spectator camera code) that the shotgun can cause. I'd suggest turning this into an early return instead.

Comment (Fueredoriku) [reply Assets/Scripts/Gamestate/PlayerManager.cs]: I agree that it should return, but the knockback issue stems from somewhere 
different (it also happens when you shoot an explosive barrel with a shotgun and get more knockback while still living)

File: Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs [hunk 2]
```diff
@@ -33,7 +36,10 @@ public void Detach(ProjectileController projectile)
     private void KnockAwayTargets(HitboxController controller, ref ProjectileState state)
     {
         Vector3 normal = gunController.transform.forward;
-
+        if (collidersHitWithShot.Item1 == state && collidersHitWithShot.Item2.Contains(controller))
```
Comment (toberge) [inline Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs:39]: note that you can actually name the fields in a tuple like `(Foo foo, Bar bar)`

File: Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs [hunk 3]
```diff
@@ -33,7 +36,10 @@ public void Detach(ProjectileController projectile)
...
+            return;
+        collidersHitWithShot.Item1 = state;
+        collidersHitWithShot.Item2.Add(controller);
```
Comment (toberge) [inline Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs:48]: When do we clear the list?

**Review by toberge — APPROVED** [review]
Seems good, though the list of hit colliders in knockbackmodifier seems to be able to grow indefinitely, perhaps it should be a set or something?

File: Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs (hunk 3 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs:48]: Ye, slight oversight, will fix

File: Assets/Scripts/Gamestate/PlayerManager.cs (hunk 1 above)
Comment (toberge) [reply Assets/Scripts/Gamestate/PlayerManager.cs]: Oh dear.

File: Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs (hunk 2 above)
Comment (toberge) [reply Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs:39]: Also I'm not sure why this isn't just two variables?

Comment (Fueredoriku) [reply Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs:39]: Could have been, but I think they should be a tupple as they are only meant to be used together

**Pull request merged**